### ISSUE TITLE
Unable to create Period to date secondary calc with period as day when metric grain is day

### ISSUE THREAD

**Issue by nipunj15** [dbt-labs/dbt_metrics#30]
The following metric fails. Using the dbt-bigquery adaptor.
Metric mode file named order_count_daily.sql
```
-- depends_on: {{ ref('enriched_orders') }}
{{ config(materialized = 'table') }}

select *
from {{ metrics.metric(
    metric_name='order_count_daily',
    grain='day',
    dimensions=[],
    secondary_calculations=[
        metrics.period_to_date(aggregate="sum", period="day", alias="pod_sum_month")
    ]
) }}

```
Yaml file containing metric definition
```
metrics:
- name: order_count_daily
  label: Order count daily
  model: ref('enriched_orders')
  description: Count of orders for each day
  type: count
  sql: order_id
  timestamp: order_date
  time_grains: [day]
  filters: []
  dimensions: []
```
Error - 
```
10:56:28  Database Error in model order_count_daily (models/mascot/order_count_daily.sql)
10:56:28    Name date_day is ambiguous inside spine at [69:67]
10:56:28    compiled SQL at target/run/jaffle_shop/models/mascot/order_count_daily.sql
```
The same secondary calculation works if Metric grain is `month` & period is `month`

Compiled sql looks like so 
```


  create or replace table `badger-345908`.`transformed`.`order_count_daily`
  
  
  OPTIONS()
  as (
    -- depends_on: `badger-345908`.`transformed`.`enriched_orders`






select *
from -- Need this here, since the actual ref is nested within loops/conditions:
    -- depends on: `badger-345908`.`transformed`.`dbt_metrics_default_calendar`
    (with source_query as (

    select
        /* Always trunc to the day, then use dimensions on calendar table to achieve the _actual_ desired aggregates. */
        /* Need to cast as a date otherwise we get values like 2021-01-01 and 2021-01-01T00:00:00+00:00 that don't join :( */
        cast(timestamp_trunc(
        cast(cast(order_date as date) as timestamp),
        day
    ) as date) as date_day,
        
        order_id as property_to_aggregate

    from `badger-345908`.`transformed`.`enriched_orders`
    where 1=1
),

 spine__time as (
     select 
        /* this could be the same as date_day if grain is day. That's OK! 
        They're used for different things: date_day for joining to the spine, period for aggregating.*/
        date_day as period, 
        
            date_day,
        
        
        date_day
     from `badger-345908`.`transformed`.`dbt_metrics_default_calendar`
 ),

spine as (

    select *
    from spine__time

),

joined as (
    select 
        spine.period,
        
        spine.date_day,
        
        

        -- has to be aggregated in this CTE to allow dimensions coming from the calendar table
    count(source_query.property_to_aggregate)
 as order_count_daily,
        logical_or(source_query.date_day is not null) as has_data

    from spine
    left outer join source_query on source_query.date_day = spine.date_day
    
    group by 1, 2

),

bounded as (
    select 
        *,
         min(case when has_data then period end) over ()  as lower_bound,
         max(case when has_data then period end) over ()  as upper_bound
    from joined 
),

secondary_calculations as (

    select *
        
        , 
    sum(order_count_daily)
over (
            partition by date_day
            
            order by period
            rows between unbounded preceding and current row
        )

as pod_sum_month

        

    from bounded
    
),

final as (
    select
        period
        
        , coalesce(order_count_daily, 0) as order_count_daily
        
        , pod_sum_month
        

    from secondary_calculations
    where period >= lower_bound
    and period <= upper_bound
    order by 1
)

select * from final

) metric_subq
  );
  ```
As can be seen, the issue is with the following CTE - where date_day column is being selected twice leading to ambiguity when selecting this column from the CTE
```
spine__time as (
     select 
        /* this could be the same as date_day if grain is day. That's OK! 
        They're used for different things: date_day for joining to the spine, period for aggregating.*/
        date_day as period, 
        
            date_day,
        
        
        date_day
     from `badger-345908`.`transformed`.`dbt_metrics_default_calendar`
 ),
```

Although it doesn't make much sense to create a period to date secondary calc. with period as day if the metric grain is day (since the metric as well as secondary calc will be same), I feel the model should not fail & the behaviour should be consistent across granularities

**Comment by callum-mcdata** [comment]
Hey @nipunj15 ! Thanks for filing this one - I've added it to my backlog and will take a look at this behavior. Love the detailed issue! From only reading your descriptions, I definitely agree that this should work and sounds like a bug.

**Comment by callum-mcdata** [comment]
Hey @nipunj15 ! This has been resolved with the v0.3.0 release. Additionally I've created a PR that adds this specific set of circumstances to a test #71 to ensure that it works in the future. 

As it has been confirmed to work and test to be merged this week, I'm closing out this issue. Thanks for flagging!
